orc/src/main/java/org/apache/iceberg/orc/OrcFileAppender.java

```diff
@@ -99,9 +110,27 @@ public Metrics metrics() {
 
   @Override
   public long length() {
-    Preconditions.checkState(isClosed,
-        "Cannot return length while appending to an open file.");
-    return file.toInputFile().getLength();
+    if (isClosed) {
+      return file.toInputFile().getLength();
+    }
+    if (this.treeWriter == null) {
+      throw new RuntimeException("Can't get the length!");
+    }
+    long estimateMemory = this.treeWriter.estimateMemory();
```

Comment:
In iceberg, we usually don't use `this.` to access the private member because we expect to use `this` to distingush  the private member and local variable when assigning a given value to someone.